src/lib/pubkey/curve448/curve448.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+* Curve448
+* (C) 2024 Jack Lloyd
+*     2024 Fabian Albert - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/curve448.h>
+
+#include <botan/ber_dec.h>
+#include <botan/der_enc.h>
+#include <botan/rng.h>
+#include <botan/internal/curve448_internal.h>
+#include <botan/internal/pk_ops_impl.h>
+
+namespace Botan {
+
+namespace {
+void curve448_basepoint_from_data(std::span<uint8_t, 56> mypublic, std::span<const uint8_t, 56> secret) {
+   auto bp = x448_basepoint(decode_scalar(secret));
+   auto bp_bytes = encode_point(bp);
+   copy_mem(mypublic, bp_bytes);
```

Comment:
If `encode_point()` were to take an out-param, we could avoid the copy. (No strong opinion, as this isn't a lot of data, but given that `curve448_basepoint_from_data()` takes an out-param it would be consistent.)